### ISSUE TITLE
--with-native-compilation

### ISSUE THREAD

**Issue by borkdude** [caldwell/build-emacs#107]
Thanks a lot for providing emacs for osx! 

Any change we can get builds with `--with-native-compilation` enabled?

**Comment by mdbergmann** [comment]
AFAIK this default in Emacs master branch? So any nightly will have it.

**Comment by sfdc-slumos** [comment]
Apparently not.
```
error: Emacs was not compiled with native compiler support (--with-native-compilation)
```
Downloaded [Emacs-pretest-28.0.92-universal.dmg](https://emacsformacosx.com/emacs-builds/Emacs-pretest-28.0.92-universal.dmg), got error, came here, downloaded [Emacs-2022-03-20_00-09-17-9b34005…-universal.dmg](https://emacsformacosx.com/emacs-builds/Emacs-2022-03-20_00-09-17-9b34005c32462a3e0e8a201e3b0d52a67a84d7bd-universal.dmg) same error. So that was fun...

**Comment by maikol-solis** [comment]
👍

**Comment by caldwell** [comment]
I've worked on it a little so far but it's harder than you'd expect to actually ship it:

1. Native comp requires libgccjit, which means compiling all of gcc.
2. The resulting binary expects `ld` and `as` to be there when it runs, and complains loudly when they are not. So that means either emacs suddenly needs xcode installed, or I have to figure out if I can ship gas and gnu ld.

I've done 1, it wasn't terribly difficult. But I need to figure out 2.

**Comment by ditdb** [comment]
I anticipate the feature very much！

**Comment by claytonrcarter** [comment]
*Asking w/ no knowledge of what is actually involved...*

@caldwell Is it an option to disable native comp support by default, but then enable it at runtime if the host system has xcode/`ld`/`as`/etc installed? Not an ideal solution, but might it be a way forward until a better solution is found?

**Comment by e40** [comment]
> I've worked on it a little so far but it's harder than you'd expect to actually ship it:
> 
>     1. Native comp requires libgccjit, which means compiling all of gcc.
> 
>     2. The resulting binary expects `ld` and `as` to be there when it runs, and complains loudly when they are not. So that means either emacs suddenly needs xcode installed, or I have to figure out if I can ship gas and gnu ld.
> 
> 
> I've done 1, it wasn't terribly difficult. But I need to figure out 2.

What about requiring the Command Line xcode tools to be installed?  That would remove the responsibility from your side... just a thought.

**Comment by denisw** [comment]
Has this become easier to accomplish now that the build is performed using Nix (https://emacsformacosx.com/about)?

**Comment by ahihi** [comment]
speaking of Nix, the `emacsPgtk` package (and possibly others) at [nix-community/emacs-overlay](https://github.com/nix-community/emacs-overlay) comes with native-comp and is working great for me on macOS 10.14. perhaps something to take inspiration from? :)

**Comment by alurm** [comment]
Seems like native compilation is enabled now? So maybe this can be closed? `(native-comp-available-p)` evaluates to `t` for me on latest emacsformacosx.

**Comment by caldwell** [comment]
Good call. https://emacsformacosx.com/about#native-comp has more info. 31.1 and 30.2-2 are the 2 releases (so far) that have native comp.
